core-api/src/test/java/com/optimizely/ab/OptimizelyTest.java

```diff
@@ -2636,6 +2638,309 @@ public void getVariationWithInvalidDatafile() throws Exception {
 
     //======== Notification listeners ========//
 
+    boolean isListenerCalled = false;
```

Comment:
This is not a safe way to determine if the listener got called. If we wanted to have a class method then it's lifecycle should be handled via `@Before` and `@After` methods. But in this case we should add a counter to the `DecisionNotificationListener` implementation. Ideally, in the form of an `AtomicInteger` that initializes to zero and increments every time `onDecision` is called. Then within each test we instantiate an instance of this listener, then assert on the value of the counter.